agent-install/agent-install.sh

```diff
@@ -384,21 +429,159 @@ function get_anax_release_version() {
     #else return empty string, meaning we couldn't find it (so they should probably just use latest)
 }
 
+# get the exchange AgentFileVersion object. 
+function get_agent_file_versions() {
+    log_debug "get_agent_file_versions() begin"
+
+    # make sure it is only called once
+    if [ $AGENT_FILE_VERSIONS_STATUS -ne 0 ]; then
+        return 0
+    fi
+
+    # input checking requires either user creds or node creds
+    local exch_creds cert_flag
+    if [[ -n $HZN_EXCHANGE_USER_AUTH ]]; then exch_creds="$HZN_ORG_ID/$HZN_EXCHANGE_USER_AUTH"
+    elif [[ -n $HZN_EXCHANGE_NODE_AUTH ]]; then exch_creds="$HZN_ORG_ID/$HZN_EXCHANGE_NODE_AUTH"
+    else
+        # no creds available to get the agent file versions
+        log_debug "No user or node credentials available to get the agent file versions"
+        AGENT_FILE_VERSIONS_STATUS=2
+        return 0
+    fi
+
+    if [[ -n $AGENT_CERT_FILE && -f $AGENT_CERT_FILE ]]; then
+        cert_flag="--cacert $AGENT_CERT_FILE"
+    elif [[ -f $PERMANENT_CERT_PATH ]]; then
+        # using cert from a previous install, see if that works
+        cert_flag="--cacert $PERMANENT_CERT_PATH"
+    else
+        cert_flag="--insecure"
+    fi
```

Comment:
Should there be logic to automatically retry with --insecure if the old cert does not work?

Reply:
Tried, got a lot of problems for getting the cfg file. Will leave it this way for now.